application/views.py

```
@@ -0,0 +1,105 @@
+        if form.is_valid():
+            f = form.save(commit=False)
+            f.user_id = user_id
+            f.application_number = (
+                str(user_id) + str(f.school.school_name) + str(f.program)
+            )
+            f.submitted_date = timezone.now()
+            try:
+                if "Submit" in request.POST.get("submit"):
+                    f.is_draft = False
+                else:
+                    f.is_draft = True
+            except:  # noqa: E722
+                pass
+            f.save()
+            return HttpResponseRedirect(reverse("application:index"))
+    else:
+        form = HighSchoolApplicationForm()
+    context = {"form": form}
+    return render(request, "application/application-form.html", context)
+
+
+def save_existing_application(request, application_id):
+    if request.method == "POST":
+        form = HighSchoolApplicationForm(request.POST)
+        if form.is_valid():
+            user_id = 1
+            f = HighSchoolApplication.objects.get(pk=application_id)
+            form = form.save(commit=False)
+            f.first_name = form.first_name
+            f.last_name = form.last_name
+            f.school = form.school
+            f.program = form.program
+            f.application_number = (
+                str(user_id) + str(f.school.school_name) + str(f.program)
+            )
+            f.email_address = form.email_address
+            f.phoneNumber = form.phoneNumber
+            f.address = form.address
+            f.gender = form.gender
+            f.date_of_birth = form.date_of_birth
+            f.gpa = form.gpa
+            f.parent_name = form.parent_name
+            f.parent_phoneNumber = form.parent_phoneNumber
+            f.submitted_date = timezone.now()
+            try:
+                if "Submit" in request.POST.get("submit"):
+                    f.is_draft = False
+                else:
+                    f.is_draft = True
+            except:  # noqa: E722
```

Maybe include an explicit exception catch. It generally isn't a good practice to use bare except blocks.

---

changed the condition to remove the exception